packages/mgt-chat/src/statefulClient/StatefulGraphChatListClient.ts

```diff
@@ -203,40 +203,47 @@ class StatefulGraphChatListClient implements StatefulClient<GraphChatListClient>
     await this.loadAndAppendChatThreads('', items, state.chatThreads.length + this.chatThreadsPerPage);
   }
 
-  private async loadAndAppendChatThreads(nextLink: string, items: GraphChatThread[], maxItems: number) {
-    if (maxItems < 1) {
-      error('maxItem is invalid: ' + maxItems);
+  private async handleChatThreadsResponse(
+    latestChatThreads: ChatThreadCollection,
+    items: GraphChatThread[],
+    maxItems: number
+  ) {
+    const latestItems = latestChatThreads.value as GraphChatThread[];
+    const checkedItems = await this.checkWhetherToMarkAsRead(latestItems);
+
+    const idsIncheckedItems = new Set(checkedItems.map(item => item.id));
+    items = items.filter(item => !idsIncheckedItems.has(item.id));
+    items = items.concat(checkedItems);
```

Comment:
This is going to re-order items. At a minimum we need to sort after doing this.

Reply:
done